perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/VosManagerBlImpl.java

```diff
@@ -385,6 +400,52 @@ public Vo getVoById(PerunSession sess, int id) throws InternalErrorException, Vo
 		}
 	}
 
+	@Override
+	public List<MemberCandidate> getCompleteCandidates(PerunSession sess, Vo vo, List<String> attrNames, String searchString) throws InternalErrorException {
+		List<ExtSource> extSources = getPerunBl().getExtSourcesManagerBl().getVoExtSources(sess, vo);
+
+		return this.getCompleteCandidates(sess, vo, null, attrNames, searchString, extSources);
+	}
+
+	@Override
+	public List<MemberCandidate> getCompleteCandidates(PerunSession sess, Group group, List<String> attrNames, String searchString, List<ExtSource> extSources) throws InternalErrorException {
+		Vo vo = getPerunBl().getGroupsManagerBl().getVo(sess, group);
+
+		return this.getCompleteCandidates(sess, vo, group, attrNames, searchString, extSources);
+	}
+
+	/**
+	 * Finds MemberCandidates who can join the Group or Vo if the Group is null.
+	 *
+	 * @param sess session
+	 * @param vo vo to be used
+	 * @param group group to be used
+	 * @param attrNames attrNames
+	 * @param searchString depends on the extSource of the Vo, could by part of the name, email or something like that.
+	 * @param extSources extSources used to find candidates
+	 * @return list of memberCandidates who match the searchString
+	 * @throws InternalErrorException internal error
+	 */
+	private List<MemberCandidate> getCompleteCandidates(PerunSession sess, Vo vo, Group group, List<String> attrNames, String searchString, List<ExtSource> extSources) throws InternalErrorException {
+		List<Candidate> candidates;
+
+		if (group != null) {
+			candidates = findCandidates(sess, group, searchString, extSources, false);
+		} else {
+			candidates = findCandidates(sess, vo, searchString, 0, extSources, false);
+		}
+
+		List<RichUser> users;
+
+		try {
+			users = getPerunBl().getUsersManagerBl().findRichUsersWithAttributes(sess, searchString, attrNames);
```

Comment:
Based on RPC API javadoc for group manager it should search only through members of VO, but this method search through whole perun.

Reply:
Now it should search only users in Vo, if it is called by group admin.